Some weird build errors on macOS AppleClang 9/SDK 10.13 (Darling XCode)

It didn't happen with the `0.55.2` tag, so it's a regression.

```
[ 21%] Building CXX object CMakeFiles/engine-lib.dir/GeneratedSource/DaemonBuildInfo/Engine.cpp.o
[ 21%] Building CXX object CMakeFiles/engine-lib.dir/src/common/Color.cpp.o
[ 22%] Building CXX object CMakeFiles/engine-lib.dir/src/common/Command.cpp.o
[ 22%] Building CXX object CMakeFiles/engine-lib.dir/src/common/Cvar.cpp.o
[ 22%] Building CXX object CMakeFiles/engine-lib.dir/src/common/Debugger.cpp.o
[ 23%] Building CXX object CMakeFiles/engine-lib.dir/src/common/FileSystem.cpp.o
[ 23%] Building CXX object CMakeFiles/engine-lib.dir/src/common/IPC/CommandBuffer.cpp.o
[ 23%] Building CXX object CMakeFiles/engine-lib.dir/src/common/IPC/Primitives.cpp.o
[ 24%] Building CXX object CMakeFiles/engine-lib.dir/src/common/KeyIdentification.cpp.o
[ 24%] Building CXX object CMakeFiles/engine-lib.dir/src/common/LineEditData.cpp.o
[ 24%] Building CXX object CMakeFiles/engine-lib.dir/src/common/Log.cpp.o
[ 25%] Building CXX object CMakeFiles/engine-lib.dir/src/common/String.cpp.o
In file included from /Volumes/SystemRoot/Unvanquished/daemon/src/common/FileSystem.cpp:32:
In file included from /Volumes/SystemRoot/Unvanquished/daemon/libs/minizip/unzip.h:51:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/zlib.h:663:46: error: expected function body after function declarator
                                             __OSX_AVAILABLE_STARTING(__MAC_10_13,__IPHONE_11_0);
                                             ^
/usr/include/Availability.h:200:50: note: expanded from macro '__OSX_AVAILABLE_STARTING'
    #define __OSX_AVAILABLE_STARTING(_osx, _ios) __AVAILABILITY_INTERNAL##_osx
                                                 ^
<scratch space>:49:1: note: expanded from here
__AVAILABILITY_INTERNAL__MAC_10_13
^
In file included from /Volumes/SystemRoot/Unvanquished/daemon/src/common/FileSystem.cpp:32:
In file included from /Volumes/SystemRoot/Unvanquished/daemon/libs/minizip/unzip.h:51:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/zlib.h:1295:37: error: expected function body after function declarator
                                    __OSX_AVAILABLE_STARTING(__MAC_10_13,__IPHONE_11_0);
                                    ^
/usr/include/Availability.h:200:50: note: expanded from macro '__OSX_AVAILABLE_STARTING'
    #define __OSX_AVAILABLE_STARTING(_osx, _ios) __AVAILABILITY_INTERNAL##_osx
                                                 ^
<scratch space>:49:1: note: expanded from here
__AVAILABILITY_INTERNAL__MAC_10_13
^
In file included from /Volumes/SystemRoot/Unvanquished/daemon/src/common/FileSystem.cpp:32:
In file included from /Volumes/SystemRoot/Unvanquished/daemon/libs/minizip/unzip.h:51:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/zlib.h:1434:38: error: expected function body after function declarator
                                     __OSX_AVAILABLE_STARTING(__MAC_10_13,__IPHONE_11_0);
                                     ^
/usr/include/Availability.h:200:50: note: expanded from macro '__OSX_AVAILABLE_STARTING'
    #define __OSX_AVAILABLE_STARTING(_osx, _ios) __AVAILABILITY_INTERNAL##_osx
                                                 ^
<scratch space>:49:1: note: expanded from here
__AVAILABILITY_INTERNAL__MAC_10_13
^
In file included from /Volumes/SystemRoot/Unvanquished/daemon/src/common/FileSystem.cpp:32:
In file included from /Volumes/SystemRoot/Unvanquished/daemon/libs/minizip/unzip.h:51:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/zlib.h:1469:39: error: expected function body after function declarator
                                      __OSX_AVAILABLE_STARTING(__MAC_10_13,__IPHONE_11_0);
                                      ^
/usr/include/Availability.h:200:50: note: expanded from macro '__OSX_AVAILABLE_STARTING'
    #define __OSX_AVAILABLE_STARTING(_osx, _ios) __AVAILABILITY_INTERNAL##_osx
                                                 ^
<scratch space>:49:1: note: expanded from here
__AVAILABILITY_INTERNAL__MAC_10_13
^
In file included from /Volumes/SystemRoot/Unvanquished/daemon/src/common/FileSystem.cpp:32:
In file included from /Volumes/SystemRoot/Unvanquished/daemon/libs/minizip/unzip.h:51:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/zlib.h:1729:37: error: expected function body after function declarator
                                    __OSX_AVAILABLE_STARTING(__MAC_10_13,__IPHONE_11_0);
                                    ^
/usr/include/Availability.h:200:50: note: expanded from macro '__OSX_AVAILABLE_STARTING'
    #define __OSX_AVAILABLE_STARTING(_osx, _ios) __AVAILABILITY_INTERNAL##_osx
                                                 ^
<scratch space>:49:1: note: expanded from here
__AVAILABILITY_INTERNAL__MAC_10_13
^
In file included from /Volumes/SystemRoot/Unvanquished/daemon/src/common/FileSystem.cpp:32:
In file included from /Volumes/SystemRoot/Unvanquished/daemon/libs/minizip/unzip.h:51:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/zlib.h:1765:35: error: expected function body after function declarator
                                  __OSX_AVAILABLE_STARTING(__MAC_10_13,__IPHONE_11_0);
                                  ^
/usr/include/Availability.h:200:50: note: expanded from macro '__OSX_AVAILABLE_STARTING'
    #define __OSX_AVAILABLE_STARTING(_osx, _ios) __AVAILABILITY_INTERNAL##_osx
                                                 ^
<scratch space>:49:1: note: expanded from here
__AVAILABILITY_INTERNAL__MAC_10_13
^
In file included from /Volumes/SystemRoot/Unvanquished/daemon/src/common/FileSystem.cpp:32:
In file included from /Volumes/SystemRoot/Unvanquished/daemon/libs/minizip/unzip.h:51:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/zlib.h:1918:53: error: expected function body after function declarator
                                                    __OSX_AVAILABLE_STARTING(__MAC_10_13,__IPHONE_11_0);
                                                    ^
/usr/include/Availability.h:200:50: note: expanded from macro '__OSX_AVAILABLE_STARTING'
    #define __OSX_AVAILABLE_STARTING(_osx, _ios) __AVAILABILITY_INTERNAL##_osx
                                                 ^
<scratch space>:49:1: note: expanded from here
__AVAILABILITY_INTERNAL__MAC_10_13
^
In file included from /Volumes/SystemRoot/Unvanquished/daemon/src/common/FileSystem.cpp:32:
In file included from /Volumes/SystemRoot/Unvanquished/daemon/libs/minizip/unzip.h:51:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/zlib.h:1920:53: error: expected function body after function declarator
                                                    __OSX_AVAILABLE_STARTING(__MAC_10_13,__IPHONE_11_0);
                                                    ^
/usr/include/Availability.h:200:50: note: expanded from macro '__OSX_AVAILABLE_STARTING'
    #define __OSX_AVAILABLE_STARTING(_osx, _ios) __AVAILABILITY_INTERNAL##_osx
                                                 ^
<scratch space>:49:1: note: expanded from here
__AVAILABILITY_INTERNAL__MAC_10_13
^
[ 25%] Building CXX object CMakeFiles/engine-lib.dir/src/common/System.cpp.o
[ 25%] Building CXX object CMakeFiles/engine-lib.dir/src/common/Util.cpp.o
[ 25%] Building CXX object CMakeFiles/engine-lib.dir/src/common/cm/cm_load.cpp.o
[ 26%] Building CXX object CMakeFiles/engine-lib.dir/src/common/cm/cm_patch.cpp.o
[ 26%] Building CXX object CMakeFiles/engine-lib.dir/src/common/cm/cm_plane.cpp.o
[ 26%] Building CXX object CMakeFiles/engine-lib.dir/src/common/cm/cm_polylib.cpp.o
[ 27%] Building CXX object CMakeFiles/engine-lib.dir/src/common/cm/cm_test.cpp.o
[ 27%] Building CXX object CMakeFiles/engine-lib.dir/src/common/cm/cm_trace.cpp.o
[ 27%] Building CXX object CMakeFiles/engine-lib.dir/src/common/cm/cm_trisoup.cpp.o
In file included from /Volumes/SystemRoot/Unvanquished/daemon/src/common/FileSystem.cpp:60:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/mach-o/dyld.h:31:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/mach-o/loader.h:47:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/mach/machine/thread_status.h:33:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/mach/i386/thread_status.h:198:9: error: unknown type name '_STRUCT_X86_AVX512_STATE32'
typedef _STRUCT_X86_AVX512_STATE32 x86_avx512_state32_t;
        ^
[ 28%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/qcommon/q_math.cpp.o
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/mach/i386/thread_status.h:233:9: error: unknown type name '_STRUCT_X86_AVX512_STATE64'
typedef _STRUCT_X86_AVX512_STATE64 x86_avx512_state64_t;
        ^
[ 28%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/qcommon/q_shared.cpp.o
[ 28%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/qcommon/q_unicode.cpp.o
[ 29%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/Application.cpp.o
[ 29%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/BaseCommands.cpp.o
[ 29%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/CommandBufferHost.cpp.o
[ 30%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/CommandSystem.cpp.o
[ 30%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/CommonVMServices.cpp.o
[ 30%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/ConsoleHistory.cpp.o
[ 30%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/ConsoleField.cpp.o
[ 31%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/CrashDump.cpp.o
[ 31%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/CvarSystem.cpp.o
[ 31%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/LogSystem.cpp.o
[ 32%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/Resource.cpp.o
[ 32%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/System.cpp.o
[ 32%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/VirtualMachine.cpp.o
[ 33%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/Crypto.cpp.o
[ 33%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/Rcon.cpp.o
/Volumes/SystemRoot/Unvanquished/daemon/src/engine/framework/CommonVMServices.cpp:42:34: warning: unknown warning group '-Wunused-lambda-capture', ignored [-Wunknown-warning-option]
#pragma clang diagnostic ignored "-Wunused-lambda-capture"
                                 ^
[ 33%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/framework/Network.cpp.o
[ 34%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/qcommon/md5.cpp.o
[ 34%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/sys/con_common.cpp.o
10 errors generated.
[ 34%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/sys/con_tty.cpp.o
[ 35%] Building CXX object CMakeFiles/engine-lib.dir/src/engine/sys/con_curses.cpp.o
make[2]: *** [CMakeFiles/engine-lib.dir/src/common/FileSystem.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /Volumes/SystemRoot/Unvanquished/daemon/src/engine/framework/VirtualMachine.cpp:42:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/spawn.h:144:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/mach/exception_types.h:178:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/mach/thread_status.h:76:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/mach/machine/thread_status.h:33:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/mach/i386/thread_status.h:198:9: error: unknown type name '_STRUCT_X86_AVX512_STATE32'
typedef _STRUCT_X86_AVX512_STATE32 x86_avx512_state32_t;
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.13.sdk/usr/include/mach/i386/thread_status.h:233:9: error: unknown type name '_STRUCT_X86_AVX512_STATE64'
typedef _STRUCT_X86_AVX512_STATE64 x86_avx512_state64_t;
        ^
2 errors generated.
make[2]: *** [CMakeFiles/engine-lib.dir/src/engine/framework/VirtualMachine.cpp.o] Error 1
1 warning generated.
make[1]: *** [CMakeFiles/engine-lib.dir/all] Error 2
make: *** [all] Error 2
```

--- a/build-release
+++ b/build-release
@@ -456,7 +456,7 @@ build () {
 	mkdir -pv "${target_build_dir}"
 	mkdir -pv "${release_dir}"
 
-	local cmake_opts='-DBUILD_GAME_NATIVE_DLL=OFF -DBUILD_GAME_NATIVE_EXE=OFF'
+	local cmake_opts='-DBUILD_GAME_NATIVE_DLL=OFF -DBUILD_GAME_NATIVE_EXE=OFF -DUSE_PRECOMPILED_HEADER=OFF'
 	local cmake_cflags=''
 
 	if "${system_macos}"

